runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineJob.java

```diff
@@ -257,6 +276,19 @@ State waitUntilFinish(
         backoff.reset();
         // Check if the job is done.
         if (state.isTerminal()) {
+          if (state == State.UPDATED) {
+            LOG.info("Job {} has been updated and is running as the new job with id {}."
+                + "To access the updated job on the Dataflow monitoring console, "
+                + "please navigate to {}",
+                getJobId(),
+                getReplacedByJob().getJobId(),
+                MonitoringUtil.getJobMonitoringPageURL(
+                    getReplacedByJob().getProjectId(), getReplacedByJob().getJobId()));
+          } else if (state == State.CANCELLED) {
+            LOG.info(String.format("Job %s cancelled by user", getJobId()));
```

Comment:
Don't use String.format`. Instead Use `LOG.info("Job {} cancelled by user", getJobId())`

Also -- why are we telling the user the job was cancelled by the user? Maybe "Job {} finished with status CANCELLED"?